src/store/plugins/2d-to-3d-management.plugin.js

```diff
@@ -0,0 +1,20 @@
+import { DEFAULT_PROJECTION } from '@/config'
+import { WEBMERCATOR } from '@/utils/coordinates/coordinateSystems'
+
+/**
+ * Plugin to switch to WebMercator coordinate system when we go 3D, and swap back to the default
+ * projection when 2D is re-activated
+ *
+ * @param {Vuex.Store} store
+ */
+export default function from2Dto3Dplugin(store) {
+    store.subscribe((mutation, state) => {
+        if (DEFAULT_PROJECTION.epsg !== WEBMERCATOR.epsg && mutation.type === 'setShowIn3d') {
+            if (state.ui.showIn3d) {
+                store.dispatch('setProjection', WEBMERCATOR)
+            } else {
+                store.dispatch('setProjection', DEFAULT_PROJECTION)
+            }
+        }
```

Comment:
Here you set the projection on the store to WEBMERCATOR but in the Cesium Component the projection is hardcoded ! Doesn't seem logic. would it not make more sense to have the Cesium hardcoded to WEBMERCATOR if it need it and let the projection in the store to the on the user choosed ? I mean if in future we support more than 2 projections you will have an issue here.